task_processing/plugins/kubernetes/utils.py

```diff
@@ -81,18 +81,42 @@ def get_kubernetes_env_vars(
     return env_vars + secret_env_vars
 
 
-def get_sanitised_kubernetes_name(name: str, replace_dots: bool = False) -> str:
+def get_sanitised_kubernetes_name(
+    name: str,
+    replace_dots: bool = False,
+    replace_forward_slash: bool = False,
+    length_limit: int = 0,
+) -> str:
     """
     Helper to ensure that any names given to Kubernetes objects follow our conventions
 
     replace_dots is an optional parameter for objects such as Containers that cannot contain `.`s in
     their names (in contrast to objects such as Pods that can)
+
+    replace_forward_slash is an optional parameter for objects that may contain / in their "pretty"
+    names, but that cannot contain / in their Kubernetes name
+
+    NOTE: For names exceeding the length limit, we'll truncate the name and replace the
+    truncated portion with a unique suffix.
     """
     name = name.replace("_", "--")
     if name.startswith("--"):
         name = name.replace("--", "underscore-", 1)
     if replace_dots:
         name = name.replace(".", "dot-")
+    if replace_forward_slash:
+        name = name.replace("/", "slash-")
```

Comment:
Can we change this from `slash-` to `-slash-` and `-dot-` to make it more human readable?

Reply:
I was mostly keeping the same transforms that we do in paasta  where we do `volume_name.replace("/", "slash-").replace(".", "dot-")`

we're sorta limited on space (63 characters for docker volume names), so it's probably worth keeping these in their current "ugly" format - additionally, (imo) it's nice to be consistent with how these are named in paasta just for familiarities sake

that said, we can definitely add a `-` prefix, but the added length might cause us to truncate in some places (which is even less readable :p)

Reply:
Ah I hadn't realized this was the same as in paasta. In that case, it makes sense to be more consistent 👍